src/Compilers/CSharp/Portable/Symbols/Synthesized/ReadOnlyListType/SynthesizedReadOnlyListTypeSymbol.cs

```diff
@@ -93,6 +101,10 @@ internal static NamedTypeSymbol Create(SourceModuleSymbol containingModule, stri
             var compilation = containingModule.DeclaringCompilation;
             DiagnosticInfo? diagnosticInfo = null;
 
+            var hasReadOnlyInterfaces =
+                !compilation.IsTypeMissing(SpecialType.System_Collections_Generic_IReadOnlyCollection_T)
+                && !compilation.IsTypeMissing(SpecialType.System_Collections_Generic_IReadOnlyList_T);
```

Comment:
> compilation.IsTypeMissing

This helper doesn't seem to be actually checking if the type is missing. It's rather "IsTypeMarkedAsMissing" and therefore is not suitable for non-test code. The proper way to do this I think is to check if the return value is of type `MissingMetadataTypeSymbol`. Is that correct?

Reply:
Thanks @alrz. Logged https://github.com/dotnet/roslyn/issues/71297.